Common/AssemblyResolver.cs

```diff
@@ -59,7 +69,13 @@ public AssemblyResolver(ILogger logger, params string[] rootSearchPaths)
                 {
                     asm = Assembly.LoadFile(file);
 
-                    if (string.Equals(args.Name, asm.FullName, StringComparison.CurrentCultureIgnoreCase))
+                    if (
+                        // If the input was e.g foo.dll then compare against the file name...
+                        (isFileName && string.Equals(Path.GetFileName(asm.Location), fileName,  StringComparison.OrdinalIgnoreCase))
```

Comment:
The previous code didn't work correctly if the supplied argument was a file name.